### PR TITLE
Merge with latest upstream

### PR DESCRIPTION
**Pull request by roberto-bayardo**
The previous test that was failing is still failing, along with a new one. But the new test failure is failing in the upstream repo, so I'll keep an eye on upstream changes that might fix it.

### CODE REVIEW

**Comment by Inphi** [comment]
I'll take a look later today

File: cmd/evm/internal/t8ntool/execution.go [hunk 1]
```diff
@@ -186,16 +186,21 @@ func (pre *Prestate) Apply(vmConfig vm.Config, chainConfig *params.ChainConfig,
...
 		evm := vm.NewEVM(vmContext, txContext, statedb, chainConfig, vmConfig)
 
 		// (ret []byte, usedGas uint64, failed bool, err error)
 		msgResult, err := core.ApplyMessage(evm, msg, gaspool)
 		if err != nil {
 			statedb.RevertToSnapshot(snapshot)
-			log.Info("rejected tx", "index", i, "hash", tx.Hash(), "from", msg.From(), "error", err)
+			log.Info("rejected tx", "index", i, "hash", tx.Hash(), "from", msg.From, "error", err)
 			rejectedTxs = append(rejectedTxs, &rejectedTx{i, err.Error()})
+			gaspool.SetGas(prevGas)
```
Comment (Inphi) [inline cmd/evm/internal/t8ntool/execution.go:203]: we should reset the data gas here as well.

File: core/gaspool.go [hunk 2]
```diff
@@ -75,6 +75,16 @@ func (gp *GasPool) DataGas() uint64 {
 	return gp.dataGas
 }
 
+// SetGas sets the amount of gas with the provided number.
+func (gp *GasPool) SetGas(gas uint64) {
+	gp.gas = gas
+}
+
+// SetGas sets the amount of data gas with the provided number.
```
Comment (Inphi) [inline core/gaspool.go:83]:
```suggestion
// SetDataGas sets the amount of data gas with the provided number.
```

File: cmd/evm/internal/t8ntool/execution.go (hunk 1 above)
Comment (roberto-bayardo) [reply cmd/evm/internal/t8ntool/execution.go:204]: yep good catch, done.

**Review by Inphi — APPROVED** [review]

File: cmd/evm/internal/t8ntool/execution.go (hunk 1 above)
Comment (donatocayurin) [reply cmd/evm/internal/t8ntool/execution.go:204]: Thank you ❤️🙏

**Pull request merged**